### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by tajemouti**
## Hi Reviewer 👋 ##
Below is a list of the accomplished tasks while working on this milestone:

1. Created a feature branch for this milestone.
2. Implemented the Redux Toolkit.
3. Created individual slice files for books and categories.
4. Defined ADD, REMOVE, and CHECK action types.
5. Created booksReducers and categoriesReducer.
6. Implemented actions for adding and removing books and checking status.
7. Imported reducers from slice files to the Redux store.
8. Configured and exported the Redux store.
9. Fixed linters' errors.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  };
+
+  const reducerFunction = reducerMap[action.type];
+  return reducerFunction ? reducerFunction() : state;
+};
+
+const addBook = (book) => ({ type: ADD, book });
+const removeBook = (id) => ({ type: REMOVE, id });
+
+export { booksReducers, addBook, removeBook };
```
Comment (Gambit142) [inline src/redux/books/booksSlice.js]: - [ ] Kindly use `createSlice` from `@reduxjs/toolkit` to create the actions and the reducers in the `booksSlice`. You can read this [article](https://medium.com/geekculture/understanding-createslice-in-redux-toolkit-reactjs-eca8d20f45d7) to understand how to use `createSlice` to create the actions and reducers. 👍 😎

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    CHECK: () => 'Under construction',
+  };
+
+  const reducerFunction = reducerMap[action.type];
+  return reducerFunction ? reducerFunction() : state;
+};
+
+const checkStatus = () => ({ type: CHECK });
+
+export { categoriesReducer, checkStatus };
```
Comment (Gambit142) [inline src/redux/categories/categoriesSlice.js]: - [ ] Kindly use `createSlice` from `@reduxjs/toolkit` to create the actions and the reducers in the `categoriesSlice`. You can read this [article](https://medium.com/geekculture/understanding-createslice-in-redux-toolkit-reactjs-eca8d20f45d7) to understand how to use `createSlice` to create the actions and reducers. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @tajemouti 👋, 

Good job so far!

## Highlights
- Setup linters correctly ✔️ 
- Use of `@reduxjs/toolkit` to create the store ✔️

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by andersonlebon — APPROVED** [review]
## Hi @tajemouti 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

## To highlight :champagne: 
- Nice use of reduxtoolkit :heavy_check_mark: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**